changed trigger with range doesn't support string based quantity

This doesn't work:
(UPDATED)
```
   Scenario Outline: Changed support ranges with string quantity
    Given items:
      | type         | name       | state     |
      | Number:Power | Item_Power | <initial> |
    And a rule:
      """
      rule 'Execute rule with range conditions' do
        changed Item_Power, <conditions>
        run { |event| logger.info("Item Power: Changed from #{event.was} to #{event.state}") }
      end
      Item_Power << '<change>'
      """
    When I deploy the rule
    Then It <should> log "Item Power: Changed from <initial> to <change>" within 5 seconds
    Examples: From range
      | initial | conditions          | change | should |
      | 0 W     | from: '0 W'..'10 W' | 14 W   | should |
```

I tried this out. 

* ONLINE/OFFLINE is not defined / available. It is not a part of the default scope from openhab: https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/defaultscope/DefaultScriptScopeProvider.java

Instead, we need to use `ThingStatus::OFFLINE...ThingStatus::ONLINE`

* When #include? is used, this rule will cause:
```
Caused by: org.jruby.exceptions.TypeError: (TypeError) can't iterate from Java::OrgOpenhabCoreThing::ThingStatus
```

I guess it's because ThingStatus is not an enumerable?

* When #cover? is used, it won't match

--- a/features/quantity_type.feature
+++ b/features/quantity_type.feature
@@ -191,3 +191,26 @@ Feature:  quantity_type
       | source | unit1 | unit2 | target |
       | 0      | °C    | °F    | 32     |
       | 1      | h     | s     | 3600   |
+
+  Scenario Outline: Triggers support ranges with string quantity
+    Given items:
+      | type         | name       | state     |
+      | Number:Power | Item_Power | <initial> |
+    And a rule:
+      """
+      rule 'Execute rule with range conditions' do
+        <trigger> Item_Power, <conditions>
+        run { |event| logger.info("Item Power: <trigger>") }
+      end
+      Item_Power << '<change>'
+      """
+    When I deploy the rule
+    Then It <should> log "Item Power: <trigger>" within 5 seconds
+    Examples: From range
+      | trigger | initial | conditions          | change | should     |
+      | changed | 0 W     | from: '0 W'..'10 W' | 14 W   | should     |
+      | changed | 12 W    | from: '0 W'..'10 W' | 10 W   | should not |
+      | changed | 0 W     | to: '0 W'..'10 W'   | 14 W   | should not |
+      | changed | 0 W     | to: '0 W'..'10 W'   | 10 W   | should     |
+      | updated | 0 W     | to: '0 W'..'10 W'   | 10 W   | should     |
+      | updated | 0 W     | to: '0 W'..'10 W'   | 14 W   | should not |

--- a/lib/openhab/dsl/rules/triggers/conditions/proc.rb
+++ b/lib/openhab/dsl/rules/triggers/conditions/proc.rb
@@ -40,7 +40,7 @@ module OpenHAB
               logger.trace("Creating range proc for #{range}")
               lambda do |state|
                 logger.trace("Range proc checking if #{state} is in  #{range}")
-                range.include? state
+                range.cover? state
               end
             end
 

